sympy/solvers/ode.py

```diff
@@ -5771,7 +5771,10 @@ def _solve_undetermined_coefficients(eq, func, order, match):
 
     for i in Add.make_args(eqs):
         s = separatevars(i, dict=True, symbols=[x])
-        coeffsdict[s[x]] += s['coeff']
+        try:
+            coeffsdict[s[x]] += s['coeff']
+        except:
```

Comment:
Here the problem was when the `trialset ` was made without checking linear independency it contained all the terms and `newtrialset` had all proper terms which are linear independent after changes which i have commited `trialset` will have only valid terms so i used try and except for handling key error. any other suggestion for this?

Reply:
Bare except should not be used anywhere.

I think that this code is a bit messy because it splits different parts into different places. It would be better to write a new `_get_trialset_undetermined_coefficients(eq, func, x)` function and make sure that it fully calculates the trialset correctly in all cases including multiplying by `x` when necessary. Then that function can be used in the matching code.

Reply:
ok I will try to write a code for that function.

Reply:
![Screenshot from 2020-01-13 19-28-40](https://user-images.githubusercontent.com/44258119/72261801-8e80d680-363b-11ea-8cd6-04defbeb632b.png)
@oscarbenjamin  could you please help?  locally i am getting correct output but on travis it is failing.

Reply:
`_undetermined_coefficients_match(x**2*sin(x)*exp(x) + x*sin(x) + x, x)` this is another example where i am getting correct output on local machine but travis fails